src/builtin_string.cpp

```diff
@@ -1125,6 +1132,27 @@ static int string_split0(parser_t &parser, io_streams_t &streams, int argc, wcha
     return string_split_maybe0(parser, streams, argc, argv, true /* is_split0 */);
 }
 
+static int string_collect(parser_t &parser, io_streams_t &streams, int argc, wchar_t **argv) {
+    options_t opts;
+    opts.chomp_valid = true;
+    int optind;
+    int retval = parse_opts(&opts, &optind, 0, argc, argv, parser, streams);
+    if (retval != STATUS_CMD_OK) return retval;
+
+    auto &buff = streams.out.buffer();
+    arg_iterator_t aiter(argv, optind, streams, false);
+    while (const wcstring *arg = aiter.nextstr()) {
+        auto begin = arg->cbegin(), end = arg->cend();
+        if (opts.chomp && begin != end && arg->back() == L'\n') {
```

Comment:
`begin != end`, isn't that equivalent to `!arg->empty()`?

Reply:
It is, but we already have `begin` and `end`. Do you think `!arg->empty()` is better?

Reply:
Done

Reply:
I definitely think its dumber. This is a case where I don't want to be too clever.